be/src/storage/delta_writer.cpp

```diff
@@ -625,12 +625,14 @@ Status DeltaWriter::_build_current_tablet_schema(int64_t index_id, const POlapTa
         if (ptable_schema_param.indexes_size() > 0 && ptable_schema_param.indexes(i).has_column_param() &&
             ptable_schema_param.indexes(i).column_param().columns_desc_size() != 0 &&
             ptable_schema_param.indexes(i).column_param().columns_desc(0).unique_id() >= 0 &&
-            ptable_schema_param.version() > ori_tablet_schema->schema_version()) {
+            ptable_schema_param.version() != ori_tablet_schema->schema_version()) {
             ASSIGN_OR_RETURN(
-                    auto tablet_schema,
+                    _tablet_schema,
                     TabletSchema::create(*ori_tablet_schema, ptable_schema_param.indexes(i).schema_id(),
                                          ptable_schema_param.version(), ptable_schema_param.indexes(i).column_param()));
-            _tablet_schema = _tablet->update_max_version_schema(tablet_schema);
+            if (_tablet_schema->schema_version() > ori_tablet_schema->schema_version()) {
```

Comment:
No need to do this check?

Reply:
> No need to do this check?

I think this check is need. Because the schema version in `_tablet_schema` maybe smaller than `ori_tablet_schema`. If that, we don't need to update the tablet schema.